### PR TITLE
add spark thrift server example

### PR DESCRIPTION
**Pull request by fernandosanchezmunoz**
With Spark Thrift Server, business users can work with their shiny Business Intelligence (BI) tools, e.g. Tableau or Microsoft Excel, and connect to Apache Spark using the ODBC interface. That brings the in-memory distributed capabilities of Spark SQL’s query engine (with all the Catalyst query optimizations you surely like very much) to environments that were initially "disconnected".

Universe PR: https://github.com/mesosphere/universe/pull/1182

### CODE REVIEW

File: spark-thrift-server/1.8/README.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
+# How to use Spark Thrift Server on DC/OS
+
+[Spark Thrift Server](https://jaceklaskowski.gitbooks.io/mastering-apache-spark/content/spark-sql-thrift-server.html) Thrift JDBC/ODBC Server (aka Spark Thrift Server or STS) is Spark SQL’s port of Apache Hive’s HiveServer2 that allows JDBC/ODBC clients to execute SQL queries over JDBC and ODBC protocols on Apache Spark.
+
+With Spark Thrift Server, business users can work with their shiny Business Intelligence (BI) tools, e.g. Tableau or Microsoft Excel, and connect to Apache Spark using the ODBC interface. That brings the in-memory distributed capabilities of Spark SQL’s query engine (with all the Catalyst query optimizations you surely like very much) to environments that were initially "disconnected".
+Beside, SQL queries in Spark Thrift Server share the same SparkContext that helps further improve performance of SQL queries using the same data sources.
+
+#Dynamic Allocation
```
Comment (ryadav88) [inline spark-thrift-server/1.8/README.md]: space in between for proper markdown rendering

File: spark-thrift-server/1.9/README.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
+# How to use Spark Thrift Server on DC/OS
+
+[Spark Thrift Server](https://jaceklaskowski.gitbooks.io/mastering-apache-spark/content/spark-sql-thrift-server.html) Thrift JDBC/ODBC Server (aka Spark Thrift Server or STS) is Spark SQL’s port of Apache Hive’s HiveServer2 that allows JDBC/ODBC clients to execute SQL queries over JDBC and ODBC protocols on Apache Spark.
+
+With Spark Thrift Server, business users can work with their shiny Business Intelligence (BI) tools, e.g. Tableau or Microsoft Excel, and connect to Apache Spark using the ODBC interface. That brings the in-memory distributed capabilities of Spark SQL’s query engine (with all the Catalyst query optimizations you surely like very much) to environments that were initially "disconnected".
+Beside, SQL queries in Spark Thrift Server share the same SparkContext that helps further improve performance of SQL queries using the same data sources.
+
+#Dynamic Allocation
```
Comment (ryadav88) [inline spark-thrift-server/1.9/README.md]: space in between for proper markdown rendering

File: spark-thrift-server/1.9/README.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+The Spark Thrift Service is setup uses dynamic executor allocation by default. This requires to have an instance of the Spark Shuffle Service running.
```
Comment (ryadav88) [inline spark-thrift-server/1.9/README.md]: add a spark shuffle example and link it from here.

File: spark-thrift-server/1.8/README.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+The Spark Thrift Service is setup uses dynamic executor allocation by default. This requires to have an instance of the Spark Shuffle Service running.
```
Comment (ryadav88) [inline spark-thrift-server/1.8/README.md]: add a spark shuffle example and link it from here.

File: spark-thrift-server/1.8/README.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+```bash
+spark-thift-server.marathon.l4lb.thisdcos.directory:9000
+```
+
+If "external access" is selected (as it is by default), the service will also be available in port 9000 of the public node if running Marathon-LB. If your public node is being firewalled, please make sure to allow traffic to port 9000 (or whichever port you select in Advanced Install) to it.
+
+## Usage
+
+You can use the Spark Thrift Server from workloads running inside the cluster of outside of it (if "external access" was left selected). This example shows to to use the Beeline JDBC client to check the service correct functioning.
```
Comment (ryadav88) [inline spark-thrift-server/1.8/README.md]: of outside -> or outside

Comment (ryadav88) [inline spark-thrift-server/1.8/README.md]: *how to use the Beeline...

File: spark-thrift-server/1.8/README.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+May 21, 2017 3:53:22 PM org.apache.hive.jdbc.HiveConnection openTransport
+INFO: Will try to open client transport with JDBC Uri: jdbc:hive2://34.210.208.43:9000/default
+Connected to: Spark SQL (version 2.1.0)                       <----- THIS MEANS CONNECTION SUCCEEDED
+Driver: Hive JDBC (version 1.2.0)
+Transaction isolation: TRANSACTION_REPEATABLE_READ
+Beeline version 1.2.0 by Apache Hive
+0: jdbc:hive2://34.210.208.43:9000/default>
+```
+
+This means the Spark Thrift Server is now ready to use by any JDBC/ODBC driver.
```
Comment (ryadav88) [inline spark-thrift-server/1.8/README.md]: please add the usage instructions

File: spark-thrift-server/1.8/README.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+```bash
+docker run --rm -ti --net=host --entrypoint=./beeline.sh sutoiku/beeline:hive-1.2.0 -u jdbc:hive2://spark-thrift-server.marathon.l4lb.thisdcos.directory:9000/default
+
+```
+
+Either way, once the docker image has been downloaded, you should see an output similar to the following
+
+```bash
+$ docker run --rm -ti --net=host --entrypoint=./beeline.sh sutoiku/beeline:hive-1.2.0 -u jdbc:hive2://$PUBLIC_NODE_IP:9000/default
```
Comment (ryadav88) [inline spark-thrift-server/1.8/README.md]: I get `java.net.ConnectException: Connection refused (Connection refused) (state=08S01,code=0)`
using marathon-lb.

Comment (fernandosanchezmunoz) [inline spark-thrift-server/1.8/README.md]: This has been tested successfully for 1.8 (customer) and 1.9 (myself) with Marathon-LB. There is no reason why 1.8 should be any different, neither with the MLB connection nor with the package itself. Can you please re-test?

File: spark-thrift-server/1.8/README.md (hunk 6 above)
Comment (fernandosanchezmunoz) [inline spark-thrift-server/1.8/README.md]: this has been added above for Beeline. What's missing?

**Comment by fernandosanchezmunoz** [comment]
@ryadav88 this should be good to go. Spark shuffle example is here: https://github.com/dcos/examples/pull/175

File: spark-thrift-server/1.8/README.md (hunk 1 above)
Comment (fernandosanchezmunoz) [inline spark-thrift-server/1.8/README.md]: done

File: spark-thrift-server/1.8/README.md (hunk 4 above)
Comment (fernandosanchezmunoz) [inline spark-thrift-server/1.8/README.md]: done

File: spark-thrift-server/1.8/README.md (hunk 5 above)
Comment (fernandosanchezmunoz) [inline spark-thrift-server/1.8/README.md]: done

**Pull request merged**